src/ui/common/state/FinalityProviderBsnState.tsx

```diff
@@ -195,18 +216,80 @@ export function FinalityProviderBsnState({ children }: PropsWithChildren) {
     );
   }, []);
 
-  const isRowSelectable = useCallback((row: FinalityProvider) => {
-    return (
-      row.state === FinalityProviderStateEnum.ACTIVE ||
-      row.state === FinalityProviderStateEnum.INACTIVE
-    );
-  }, []);
+  const isRowSelectable = useCallback(
+    (row: FinalityProvider) => {
+      const statusAllowed =
+        row.state === FinalityProviderStateEnum.ACTIVE ||
+        row.state === FinalityProviderStateEnum.INACTIVE;
+
+      // For selection (not filtering), only restrict based on allowlist for non-Babylon BSNs
+      const allowlistAllowed =
+        !selectedBsnId ||
+        selectedBsnId === BBN_CHAIN_ID ||
+        Boolean(row.isAllowlisted);
+
+      return statusAllowed && allowlistAllowed;
+    },
+    [selectedBsnId],
+  );
+
+  const selectedBsn = useMemo(
+    () => bsnList.find((bsn) => bsn.id === selectedBsnId),
+    [bsnList, selectedBsnId],
+  );
 
   const filteredFinalityProviders = useMemo(() => {
-    return finalityProviders.filter((fp: FinalityProvider) =>
-      Object.values(FILTERS).every((filterFn) => filterFn(fp, filter)),
+    // First apply search filter
+    let filtered = finalityProviders.filter((fp: FinalityProvider) =>
+      FILTERS.searchTerm(fp, filter),
     );
-  }, [finalityProviders, filter]);
+
+    // If search is active, return all matching results without status filtering
+    if (filter.searchTerm) {
+      return filtered;
+    }
+
+    // Apply BSN-aware filtering based on provider status
+    if (filter.providerStatus) {
+      const bsnConfig = getBsnConfig(selectedBsn);
```

Comment:
Could be worth sending this logic to a method to improve readaiblity/testability later down the line